modules/react/text/lib/Text.tsx

```diff
@@ -12,65 +15,86 @@ export interface TextProps extends BoxProps {
    * ```tsx
    * <Text typeLevel="body.small">Small body text</Text>
    * ```
-  */
-  typeLevel?: `${keyof CanvasTypeHierarchy}.${'large'|'medium'|'small'}`;
+   */
+  typeLevel?: TokenName;
   /**
    * Type variant token names: `error`, `hint` or `inverse`.
    *
    * ```tsx
    * <Text variant="error" typeLevel="subtext.large">Error text</Text>
    * ```
-  */
-  variant?: keyof CanvasTypeVariants;
+   */
+  variant?: 'error' | 'hint' | 'inverse';
 }
 
-/**
- * Function returns updated props by token and variant values
-*/
-const validateProps = ({typeLevel, variant, ...props}: TextProps) => {
-  let updatedProps: any = props;
-  const tokenPropNames = [
-    'color',
-    'fontFamily',
-    'fontSize',
-    'fontWeight',
-    'letterSpacing',
-    'lineHeight',
-  ];
-
-  /*
-  If token provided it updates undefined values of token styles:
-  `fontFamily`, `fontSize`, `lineHeight`, `fontWeight`, `color`
-  by replacing them by token value
-  */
-  if (typeLevel) {
-    const [level, size] = typeLevel.split('.') as [
-      keyof CanvasTypeHierarchy,
-      'large' | 'medium' | 'small'
-    ];
-    const tokenProps = type.levels[level][size];
-
-    tokenPropNames.forEach(item => {
-      if (!props[item as keyof typeof props]) {
-        updatedProps[item] = tokenProps[item as keyof typeof tokenProps];
-      }
-    });
-  }
-
-  // If variant provided it updates color value by variant color
-  if (variant) {
-    updatedProps = {...updatedProps, ...type.variants[variant]};
-  }
-
-  return updatedProps;
-};
+const textModifiers = createModifiers({
+  typeLevel: {
+    // Title level styles
+    'title.large': createStyles({
+      ...system.type.title.large,
+      color: base.blackPepper400,
+    }),
+    'title.medium': createStyles({
```

Comment:
question: I know we had these with dot notation because that's how you could access them prior. Do you think it makes sense to change these to just `titleLarge`? Hmm that seems like a pain in the butt, just thinking outloud.

Reply:
I think dot notation is better because it looks like object property access and it's easier to split by dot and get expected result than camel case